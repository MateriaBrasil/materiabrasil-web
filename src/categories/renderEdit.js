import React from 'react'
import Dialog from '@material-ui/core/Dialog'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import ExpansionPanel from '@material-ui/core/ExpansionPanel'
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary'
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails'
import Typography from '@material-ui/core/Typography'
import ExpandMoreIcon from '@material-ui/icons/ExpandMore'
import FormGroup from '@material-ui/core/FormGroup'
import FormControlLabel from '@material-ui/core/FormControlLabel'
import Checkbox from '@material-ui/core/Checkbox'

import some from 'lodash/some'

// import withMobileDialog from '@material-ui/core/withMobileDialog'
// import { Edit } from 'croods'

import categories from './mock'

// import render from './edit/render'
// import reload from './reload'

const renderCategory = category => {
  if (category.categories) {
    const deepNested = some(category.categories, 'categories')
    return (
      <ExpansionPanel style={{ width: '100%', flex: 1 }}>
        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
          <Typography>{category.name}</Typography>
        </ExpansionPanelSummary>
        <ExpansionPanelDetails style={{ display: deepNested ? 'block' : 'flex' }}>
          {category.categories.map(renderCategory)}
        </ExpansionPanelDetails>
      </ExpansionPanel>
    )
  }
  return (
    <FormControlLabel
      control={
        <Checkbox
          checked={false}
          onChange={() => undefined}
          value={category.name}
        />
      }
      label={category.name}
    />
  )
}

export default props => routeProps => {

  return (
    <Dialog
      fullScreen={false}
      open
      aria-labelledby="responsive-dialog-title"
      onBackdropClick={() => undefined}
      onEscapeKeyDown={() => undefined}
      fullWidth
    >
      <DialogTitle id="responsive-dialog-title">
        Adicionar Categorias
      </DialogTitle>
      <DialogContent>
        <FormGroup row>{categories.map(renderCategory)}</FormGroup>
      </DialogContent>
    </Dialog>
  )
}
