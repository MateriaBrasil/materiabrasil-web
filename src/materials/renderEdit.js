import React from 'react'
import { Edit } from 'croods'

import render from './edit/render'
import renderUpdated from './edit/renderUpdated'

export default props => routeProps => {
  const { match } = routeProps
  const { id } = match.params

  return (
    <Edit
      id={id}
      name="materials"
      render={render({ ...props, ...routeProps })}
      renderUpdated={renderUpdated({ ...props, ...routeProps })}
    />
  )
}
