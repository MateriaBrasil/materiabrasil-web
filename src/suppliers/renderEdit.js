import React from 'react';
import { Edit } from 'croods';
import Redirect from 'react-router-dom/Redirect';

import render from './edit/render';

export default props => routeProps => {
  const { match } = routeProps;
  const { params } = match;
  const { id } = params;

  return (
    <Edit
      id={id}
      name="suppliers"
      render={render({ ...props, ...routeProps })}
      renderUpdated={() => (
        <Redirect
          to={`/suppliers/${props.currentUser.suppliers[0].slug}?reload=true`}
        />
      )}
    />
  );
};
