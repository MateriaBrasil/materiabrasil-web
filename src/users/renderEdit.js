import React from 'react'
import { Edit } from 'croods'
import Redirect from 'react-router-dom/Redirect'

import render from './edit/render'

export default props => routeProps => {
  const id = props.id

  return (
    <Edit
      id={id}
      name="users"
      render={render({ ...props, ...routeProps })}
      renderUpdated={() => <Redirect to="/profile" />}
    />
  )
}
