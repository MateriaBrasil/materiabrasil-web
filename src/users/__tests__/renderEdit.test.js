import React from 'react'
import renderer from 'react-test-renderer'

import renderEdit from '../renderEdit'

jest.mock('croods', () => ({
  Edit: props => (
    <div {...props}>
      Edit - {props.render()} {props.renderUpdated()}
    </div>
  ),
}))

jest.mock('react-router-dom/Redirect', () => props => (
  <div {...props}>Redirect</div>
))

jest.mock('../edit/render', () => props => info => (
  <div {...props}>Edit Form</div>
))

describe(' id', () => {
  it('renders correctly', () => {
    const props = { id: 1234 }
    const tree = renderer.create(renderEdit(props)({ foo: 'bar' })).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
