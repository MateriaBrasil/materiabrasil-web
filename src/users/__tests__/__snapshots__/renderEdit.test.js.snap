// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` id renders correctly 1`] = `
<div
  id={1234}
  name="users"
  render={[Function]}
  renderUpdated={[Function]}
>
  Edit - 
  <div
    foo="bar"
    id={1234}
  >
    Edit Form
  </div>
   
  <div
    to="/profile"
  >
    Redirect
  </div>
</div>
`;
